lib/platform/github/index.ts

```diff
@@ -107,9 +109,16 @@ export async function initPlatform({
   let discoveredGitAuthor: string;
   if (!gitAuthor) {
     userDetails = await getUserDetails(defaults.endpoint, token);
-    const userEmail = await getUserEmail(defaults.endpoint, token);
-    if (userEmail) {
-      discoveredGitAuthor = `${userDetails.name} <${userEmail}>`;
+    if (platformCommit) {
+      const userId = userDetails.id.toString();
+      const userName = userDetails.username;
+      const userEmail = `${userId}+${userName}@users.noreply.github.com`;
+      discoveredGitAuthor = `${userDetails.username} <${userEmail}>`;
```

Comment:
Document here in code why this is important. This is the author which GitHub will "automatically" use, right? So we need for the git layer to know it in order to recognize our own commits?

I think it also means that we need to document how people can transition from non-platform commits to platform commits. e.g. they'll need to add the old gitAuthor to `gitIgnoredAuthors`? Wondering if we could even do that automatically somehow.

Reply:
As an option, I've shouldn't go this way and make some changes to author mismatch handling code instead.
These lines just seemed to be quickest way to make the whole thing work.

Reply:
FYI the app doesn't call `initPlatform` so ideally it can be avoided

Reply:
Okay, will research on alternatives

Reply:
I can still work with this approach if needed. We should aim to satisfy the mismatched gitAuthor for people transitioning though

Reply:
If we want to address people who turn on the `platfromCommit` flag explicitly, we can just mention how to use `gitIgnoredAuthors`. But if we're about to make this transition transparent, some automatic handling is still needed.

Reply:
Could it be as simple as adding the userid+username email to `gitIgnoredAuthors`?

Reply:
With `gitIgnoredAuthors: ['77205328+renovate-testing@users.noreply.github.com']` my test repo transitioned smoothly from platform commits `off` to `on` and vice versa. However, when older branch is found, it will be reused no matter what `platformCommit` value is.

Reply:
I don't understand.. is there a problem with reuse?

Reply:
I don't think it's a problem. From platform API usage perspective, seems like it's actually very good. From user perspective, just merge existing ones, while all subsequent PRs will be created from the platform API.

Reply:
Sounds good, reuse depends on `rebaseWhen` settings.

Reply:
So can we adjust `initPlatform` so that it returns `gitAuthor` and optionally `gitIgnoredAuthors`?

Reply:
Okay, you already can check it out